rules/fetch-data-react/rule.md

```diff
@@ -0,0 +1,145 @@
+---
+type: rule
+archivedreason: 
+title: Do you know the best libraries to fetch data in React?
+guid: cae7be1d-2313-43e1-8cc5-cfc1e77b4bb4
+uri: fetch-data-nextjs
+created: 2023-08-08T05:24:35Z
+authors:
+- title: Harry Ross
+  url: https://www.ssw.com.au/people/harry-ross
+related:
+- fetch-data-nextjs
+---
+
+While using a regular `useEffect` to run when a component is loaded to load data from an API is super easy, it may result in unnecesary duplicate requests for data. It is best to use a library for this purpose, as not only does it solve the afforementioned issue, but also comes with a variety of other useful features such as caching, background updates, and pre-fetching. 
+
+<!--endintro-->
+
+Below is an example of a standard data fetch in React:
+
+::: bad
+```tsx
+
+const Component = () => {
+  const [data, setData] = useState({});
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    fetch("https://jsonplaceholder.typicode.com/todos/1")
+      .then(res => res.json())
+      .then(json => {
+        setData(json);
+        setLoading(false);
+      })
+  }, [])
+
+  return (
+    {loading
+      ? <> {/* Display data here */} </>
+      : <p>Loading...</p>
+    }
+  )
+}
+
+
+```
+:::
+
+This example is not ideal, as it means every time we reload this page component, or if we make the same request on another page, there will be an unneeded request made instead of pulling the data from a cache. 
+
+There are two primary options that both serve effectively the same purpose in providing developers with useful hooks for fetching data. These libraries not only give developers a wide range of other features, but also reduces the amount of boilerplate code they have to write. 
```

Comment:
```suggestion
Below are the two recommended options that both serve effectively the same purpose in providing developers with useful hooks for fetching data. These libraries not only give developers a wide range of other features, but also reduces the amount of boilerplate code they have to write. 
```

I'd suggest to use "recommended" instead of "primary" so this is more subjective (i.e. SSW's recommendation) compared to saying that "there are only minor smaller fetching libraries out there" 😄